sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -382,16 +382,13 @@ private final class GpuSemaphore() extends Logging {
   }
 
   def releaseIfNecessary(context: TaskContext): Unit = {
-    val nvtxRange = new NvtxRange("Release GPU", NvtxColor.RED)
-    try {
+    NvtxRegistry.RELEASE_GPU {
       val taskAttemptId = context.taskAttemptId()
       GpuTaskMetrics.get.updateRetry(taskAttemptId)
       val taskInfo = tasks.get(taskAttemptId)
       if (taskInfo != null) {
         taskInfo.releaseSemaphore(semaphore)
       }
-    } finally {
-      nvtxRange.close()
```

Comment:
did not look too deep but is this close no longer required ?  I don't see it moving to another place

Reply:
`close` used to be the way we'd pop the range, but now `apply` does this in its own `finally` block: https://github.com/NVIDIA/spark-rapids/pull/12035/files#diff-e0f763286eef700f97dbfa0e8e4e1af7fccfa7153ed28406663a7f9bd8572a73R31